test/e2e/dynamic_resource_e2e_test.go

```diff
@@ -3,103 +3,164 @@ package e2e
 import (
 	"strings"
 
-	"github.com/stretchr/testify/require"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
 
 	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
 	"github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/api/client/clientset/versioned"
 	"github.com/operator-framework/operator-lifecycle-manager/pkg/controller/registry"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/lib/operatorclient"
 	"github.com/operator-framework/operator-lifecycle-manager/test/e2e/ctx"
 )
 
-var _ = Describe("Dynamic Resource", func() {
-	It("resolve prometheus API", func() {
+var _ = Describe("Subscriptions create required objects from Catalogs", func() {
 
-		defer cleaner.NotifyTestComplete(GinkgoT(), true)
+	var (
+		c             operatorclient.ClientInterface
+		crc           versioned.Interface
+		dynamicClient dynamic.Interface
+		deleteOpts    *metav1.DeleteOptions
+	)
 
-		c := newKubeClient(GinkgoT())
-		crc := newCRClient(GinkgoT())
-		dynamicClient := ctx.Ctx().DynamicClient()
+	BeforeEach(func() {
+		c = newKubeClient()
+		crc = newCRClient()
+		dynamicClient = ctx.Ctx().DynamicClient()
+	})
+
+	AfterEach(func() {
+		deleteOpts = &metav1.DeleteOptions{}
+		cleaner.NotifyTestComplete(true)
+
+	})
 
-		ns, err := c.KubernetesInterface().CoreV1().Namespaces().Create(&corev1.Namespace{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: genName("ns-"),
-			},
+	Context("Given a Namespace with an OwnNamespace OperatorGroup", func() {
+		var (
+			ns  *corev1.Namespace
+			err error
```

Comment:
Does `err` need to be shared across this Context, or just ns?

Reply:
`err` variable is used in almost all of the context. There could be 2 ways of declaring it: 
1. globally declare it (in the outermost Describe block) or
2. redeclare it in every `Context`/`Describe` wherever it is used. 

Because each `It` spec calls BeforeEach (outermost to inner block), the variable gets redeclared and so there won't be any test pollution. In the above case, I have declared only once, starting in the `Context` where it gets used.

Reply:
IMO, there isn't any value in reducing the number of error identifiers that are declared. The real motivation behind declaring variables at the Context scope is so that values can be assigned to them in a shared Before node and later accessed in specs. Maybe I missed it, but I don't see that happening in this situation -- the _value_ assigned to `err` doesn't need to be shared throughout this test scope.

Another factor to consider is: how obvious will it be to a person who later modifies this test so that it doesn't always assign a value to `err` in `BeforeEach`?

Reply:
Makes sense. I have redeclared the variable every time it gets used in a `Context`